lib/Service/Avatar/SvgPortableSecureImage.php

```diff
@@ -0,0 +1,1173 @@
+			'stroke-width',
+			'systemLanguage',
+			'text-align',
+			'text-anchor',
+			'transform',
+			'typeof',
+			'unicode-bidi',
+			'vector-effect',
+			'width',
+			'x',
+			'xml:base',
+			'xml:id',
+			'xml:lang',
+			'xml:space',
+			'y',
+		],
+
+		// <title />
+		'title' => [
+			'about',
+			'buffered-rendering',
+			'class',
+			'content',
+			'datatype',
+			'display',
+			'id',
+			'image-rendering',
+			'property',
+			'rel',
+			'requiredFonts',
+			'resource',
+			'rev',
+			'role',
+			'shape-rendering',
+			'systemLanguage',
+			'text-rendering',
+			'typeof',
+			'viewport-fill',
+			'viewport-fill-opacity',
+			'visibility',
+			'xml:base',
+			'xml:id',
+			'xml:lang',
+			'xml:space',
+		],
+
+		// <use />
+		'use' => [
+			'about',
+			'class',
+			'color',
+			'color-rendering',
+			'content',
+			'datatype',
+			'direction',
+			'display-align',
+			'fill',
+			'fill-opacity',
+			'fill-rule',
+			'font-family',
+			'font-size',
+			'font-style',
+			'font-variant',
+			'font-weight',
+			'href',
+			'id',
+			'line-increment',
+			'property',
+			'rel',
+			'requiredFonts',
+			'resource',
+			'rev',
+			'role',
+			'solid-color',
+			'solid-opacity',
+			'stop-color',
+			'stop-opacity',
+			'stroke',
+			'stroke-dasharray',
+			'stroke-dashoffset',
+			'stroke-linecap',
+			'stroke-linejoin',
+			'stroke-miterlimit',
+			'stroke-opacity',
+			'stroke-width',
+			'systemLanguage',
+			'text-align',
+			'text-anchor',
+			'transform',
+			'typeof',
+			'unicode-bidi',
+			'vector-effect',
+			'x',
+			'xml:base',
+			'xml:id',
+			'xml:lang',
+			'xml:space',
+			'y',
+		],
+	];
+
+	/**
+	 * @var string
+	 */
+	private $data;
+
+	/**
+	 * @param string $data
+	 */
+	public function __construct(string $data) {
+		$this->data = $data;
+	}
+
+	/**
+	 * Returns true if SVG image conforms
+	 * to SVG Portable/Secure format, otherwise false.
+	 *
+	 * @return bool
+	 */
+	public function isValid(): bool {
+		return $this->suppressLibXmlErrors(function () {
+			$doc = new SimpleXMLElement($this->data);
+			$elements = $doc->xpath('descendant-or-self::*');
+
+			foreach ($elements as $element) {
+				$element_name = $element->getName();
+
+				if (!$this->isAllowedElementName($element_name)) {
+					return false;
+				}
+
+				foreach ($element->getNamespaces() as $ns) {
+					foreach ($element->attributes($ns) as $attr) {
+						$attr_allowed = $this->isAllowedAttributeName(
+							$attr->getName(),
+							$element_name
+						);
+
+						if (!$attr_allowed) {
+							return false;
+						}
+					}
+				}
+			}
+
+			return true;
+		});
+	}
+
+	/**
+	 * Convert the SVG image to PNG format
+	 * of size ($size by $size) pixels and returns
+	 * it as an instance of Imagick.
+	 *
+	 * @param int $size (default: 128):
+	 *		x and y dimensions of the resulting PNG image.
+	 *
+	 * @return \Imagick
+	 *
+	 * @throws \OCA\Mail\Exception\ServiceException
+	 */
+	public function toPngImage(int $size = 128): Imagick {
```

Comment:
As I mentioned [here](https://github.com/nextcloud/mail/pull/5784#discussion_r771506723) I am not sure about the right dimensions of the avatars.